auth/auth/auth.py

```diff
@@ -393,14 +393,6 @@ async def create_user(request: web.Request, userdata):  # pylint: disable=unused
     hail_credentials_secret_name = body.get('hail_credentials_secret_name')
     if (hail_identity or hail_credentials_secret_name) and not is_test_deployment:
         raise web.HTTPBadRequest(text='Cannot specify an existing hail identity for a new user')
-    if hail_credentials_secret_name:
-        try:
-            k8s_cache: K8sCache = request.app['k8s_cache']
-            await k8s_cache.read_secret(hail_credentials_secret_name, DEFAULT_NAMESPACE)
-        except kubernetes_asyncio.client.rest.ApiException as e:
-            raise web.HTTPBadRequest(
-                text=f'hail credentials secret name specified but was not found in namespace {DEFAULT_NAMESPACE}: {hail_credentials_secret_name}'
-            ) from e
```

Comment:
This was initially put in as a safeguard, because a user cannot submit jobs to the namespace if their gsa key secret does not exist in that namespace, but I feel like this hinders more than helps for the case when you just want other devs to be able to look at a batch in your namespace. This will also not be relevant once we phase out keys.